### DISCUSSION TITLE
Problem running the Implicit_spring-back/DBEND_44_0002.rad

### DISCUSSION THREAD

**Discussion by yw5aj** [OpenRadioss/OpenRadioss#422] [Users]
Hi all,

In running the spring-back example [here](https://openradioss.atlassian.net/wiki/spaces/OPENRADIOSS/pages/9502721/Spring-back), everything worked OK except for the `Implicit_spring-back/DBEND_44_0002.rad`. When I ran it, the following message came out:

```

MESSAGE ID :         73

 ** ERROR IN SOLVER INPUT DECK CARD: FUNCT
```

Could you help me resolve this?

Thank you,
Shawn

**Comment by PaulAltair** [comment]
Thanks for reporting Shawn, my fault!, I accidentally deleted the first line of that example file when inserting the CC license notice

the first line of the file (after the comments, before the line 1000.01), should be /RUN/DBEND_44/2

If you insert that it should work, or I have updated the zip for the example so you can redownload if you prefer

```
# RUN 2: SPRING BACK SIMULATION BY IMPLICIT
#
# Copyright (C) 2022 Altair Engineering Inc. ("Holder")
# Model is licensed by Holder under CC BY-NC 4.0
# (https://creativecommons.org/licenses/by-nc/4.0/legalcode).
#
/RUN/DBEND_44/2
1000.01
/VERS/140
/ANIM/DT
 960.  4.
```

**Reply by yw5aj** [reply]
Thank you Paul for the prompt response! For some reasons after the edits the results became the following:
```
************************************************************************
**                                                                    **
**                                                                    **
**                         OpenRadioss Engine                         **
**                                                                    **
**            Non-linear Finite Element Analysis Software             **
**                                                                    **
**                                                                    **
**                                                                    **
**                    Linux 64 bits, GNU compiler                     **
**                                                                    **
**                                                                    **
**                                                                    **
** CommitID: 7e0fea2fce76f8d27fe526d315e9cb57c9ce2c0b                 **
************************************************************************
** OpenRadioss Software                                               **
** COPYRIGHT (C) 1986-2022 Altair Engineering, Inc.                   **
** Licensed under GNU Affero General Public License.                  **
** See License file.                                                  **
************************************************************************

 ROOT: DBEND_44  RESTART: 0002
 NUMBER OF HMPP PROCESSES     8
 10/09/2022
*** stack smashing detected ***: terminated
*** stack smashing detected ***: terminated
*** stack smashing detected ***: terminated
*** stack smashing detected ***: terminated
*** stack smashing detected ***: terminated


User or system abort detected (CTRL-C) !
*** stack smashing detected ***: terminated

*** stack smashing detected ***: terminated
*** stack smashing detected ***: terminated


User or system abort detected (CTRL-C) !



User or system abort detected (CTRL-C) !



User or system abort detected (CTRL-C) !



User or system abort detected (CTRL-C) !



User or system abort detected (CTRL-C) !



User or system abort detected (CTRL-C) !



User or system abort detected (CTRL-C) !

     ERROR TERMINATION
     ERROR TERMINATION
     ERROR TERMINATION
```

And just to be safe, my file after the edit looks like:

```
# RUN 2: SPRING BACK SIMULATION BY IMPLICIT
#
# Copyright (C) 2022 Altair Engineering Inc. ("Holder")
# Model is licensed by Holder under CC BY-NC 4.0
# (https://creativecommons.org/licenses/by-nc/4.0/legalcode).
#
/RUN/DBEND_44/2
1000.01
/VERS/140
/ANIM/DT
 960.  4.
/ANIM/VECT/DISP
/ANIM/VECT/VEL
/ANIM/VECT/FOPT
/ANIM/ELEM/EPSP
/ANIM/ELEM/VONM
/ANIM/ELEM/ENER

/ANIM/ELEM/THICK

/ANIM/ELEM/HOUR
/ANIM/SHELL/TENS

/RBODY/ON
2549    2550    2551
/TFILE
2.
#/ABF
#2. 1e30
/PRINT/-1
/RFILE
 20000 0 0
/MON/ON   

#INTERFACES REMOVING:

/DEL/INTER
1 2 3

#ADDED BOUNDARY CONDITIONS:
                                   
/BCS/TRA/XYZ/
2549 2550 
/BCS/ROT/XYZ/
2549 2550 
/BCS/TRA/Z/
427


####################
# IMPLICIT OPTIONS #
####################

/IMPL/PRINT/NONL/-1
/IMPL/SOLVER/1
# IPREC L_LIM ITOL L_TOL
 0 0 0 0.
/IMPL/NONLIN/1
# upd_K_LIM NITOL N_TOL
 2 0 0.25e-1
/IMPL/DTINI
 0.08
/IMPL/DT/STOP
# DT_MIN DT_MAX
 0.1e-4 0.0
/IMPL/DT/2
# NL_DTP SCAL_DTP NL_DTN SCAL_DTN
 6 .0 20 0.67 0.0
/IMPL/SPRBACK
```

Any thoughts on why that is happening?

**Reply by PaulAltair** [reply]
no sorry, never seen the 'stack smashing' message before and not sure where it comes from. The input file should be ok now regards syntax.

**Comment by yw5aj** [comment]
Thanks Pauly! Could be my WSL issue. Let me try a bit harder and let you
know!

Shawn

**Reply by PaulAltair** [reply]
I have now recompiled/reinstalled latest OpenRadioss in my VM and I have an issue with that 0002.rad too, I get 'Segmentation Violation' so I guess the 'stack smashing' is the WSL equivalent! I have asked someone to take a look

**Reply by MarianBulla** [reply]
Be sure if you have VERSION 2 of WSL.
You can check your WSL version with (in a command window): wsl -l -v

**Comment by yw5aj** [comment]
Phew - relieving to know that I'm not alone! :)

Thanks again for the prompt response and I look forward to hearing back.

-Shawn

**Comment by yw5aj** [comment]
Hi Marian,

Thanks for the info! Yes I am using WSL2.

Shawn

On Mon, Sep 12, 2022 at 3:45 AM Marian Bulla ***@***.***>
wrote:

> Be sure if you have VERSION 2 of WSL.
> You can check your WSL version with (in a command window): wsl -l -v
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/OpenRadioss/discussions/422#discussioncomment-3624748>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA2RB3IXO4K52TJG7F53WZ3V54CVTANCNFSM6AAAAAAQJPDTZY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Yuxiang "Shawn" Wang, PhD
***@***.***
+1 (434) 284-0836

**Comment by meiyue1989** [comment]
When I run the same case Implicit_spring-back/DBEND_44_0002.rad using WSL 2.0, I got the following error message: 
 ROOT: DBEND_44  RESTART: 0002
 NUMBER OF HMPP PROCESSES     1
 19/09/2022

MESSAGE ID :         44

 ** ERROR: ABNORMAL END
      +=============================================================+
      |          Segmentation Violation                             |
      +=============================================================+
      | TRACE BACK:                                                |
     ERROR TERMINATION

I would like to know how to address this  issue. I found when I use impl/linear, I will get the same error message.  I also got the same error message when I compiled and ran the code  in the supercomputer.

**Reply by PaulAltair** [reply]
I think this is related to the link to MUMPS being unavailable yet in the OpenRadioss release. I will update the comments on the tutorial.

**Reply by meiyue1989** [reply]
Dear Sir,
I did not use the release one, I compiled the source code in WSL 2.0 and got this issue.  I also debugged the code using gdb, and got a interrupted exception issue at 
        GBUF%IERR(1:NEL*G_IERR)  = NINT(RBUF(IAD+1:IAD+NEL*G_IERR))       at allocbuf.auto.f 
 It will not occur when I use the explicit solver. 
Best,
Yue

**Reply by yw5aj** [reply]
> I think this is related to the link to MUMPS being unavailable yet in the OpenRadioss release. I will update the comments on the tutorial.

Ah that's why! It makes sense. Being curious, do you have a rough idea on whether we plan to support that soon? It seems like without mumps we almost have no implicit capability?

Shawn

**Reply by PaulAltair** [reply]
There is a separate thread on this, https://github.com/OpenRadioss/OpenRadioss/discussions/485#discussioncomment-3727927 there is a slightly complex workaround to compile with implicit capabilities, and a plan to document the precise steps more clearly

**Reply by meiyue1989** [reply]
> There is a separate thread on this, [#485 (comment)](https://github.com/orgs/OpenRadioss/discussions/485#discussioncomment-3727927) there is a slightly complex workaround to compile with implicit capabilities, and a plan to document the precise steps more clearly

Dear Sir,
Do you mean you will provide a document to compile the source code so that it is possible to use the implict solver? If yes, that would be fantastic. 
Best,
Yue

**Reply by PaulAltair** [reply]
Yes, it should already be possible from the steps described but it is not straightforward, a step by step guide will follow

**Reply by meiyue1989** [reply]
> Yes, it should already be possible from the steps described but it is not straightforward, a step by step guide will follow

Dear Sir, 
I would like to know whether there is any update on the guide on the building the implict solver?
Best,
Yue
